lib/validate/net_test.go

```diff
@@ -73,3 +73,40 @@ func (*S) TestValidateKubernetesSubnets(c *check.C) {
 		}
 	}
 }
+
+func (*S) TestNetworkOverlap(c *check.C) {
+	type testCase struct {
+		ipAddr      string
+		subnetCIDR  string
+		ok          bool
+		description string
+	}
+	testCases := []testCase{
+		{
+			ipAddr:      "10.100.0.0",
+			subnetCIDR:  "10.100.0.0/16",
+			ok:          false,
+			description: "At the edge of the range.",
+		},
+		{
+			ipAddr:      "10.100.0.111",
+			subnetCIDR:  "10.100.0.0/16",
+			ok:          false,
+			description: "Inside the range.",
+		},
+		{
+			ipAddr:      "11.100.0.111",
+			subnetCIDR:  "10.100.0.0/16",
+			ok:          true,
+			description: "Outside the range.",
+		},
+	}
+	for _, tc := range testCases {
+		err := NetworkOverlap(tc.ipAddr, tc.subnetCIDR, "errMsg")
+		if tc.ok {
+			c.Assert(err, check.IsNil, check.Commentf(tc.description))
```

Comment:
Ideally this should verify that the err is also an expected one. You could used ErrorMatches checker instead.